src/Polly.Core/Strategy/Outcome.cs

```diff
@@ -46,6 +57,11 @@ public Outcome(object? result)
     /// </summary>
     public bool IsVoidResult => Result is VoidResult;
 
+    /// <summary>
+    /// Gets the <see cref="ExceptionDispatchInfo"/> associated with the exception, if any.
+    /// </summary>
+    internal ExceptionDispatchInfo? ExceptionDispatchInfo { get; }
```

Comment:
Given this new property, then you should change the Exception property to just

```csharp
public Exception? => ExceptionDispatchInfo?.SourceException;
```

which would shrink the Outcome and Outcome<T> structs by 1/3, speeding up all that struct copying.